### PR TITLE
Update Framerate Histogram to be a Frame Time histogram

### PR DESCRIPTION
**Pull request by rgba16f**
present average, min, max frame time in milliseconds

Signed-off-by: rgba16f <82187279+rgba16f@users.noreply.github.com>

### CODE REVIEW

File: Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp [hunk 1]
```diff
@@ -1,105 +1,112 @@
...
+        m_averageLog.reserve(m_maxSamples);
+    }
+
+    float ImGuiHistogramQueue::UpdateDisplayedValues(AZStd::size_t maxSampleCount, float& minValue, float& maxValue)
+    {
+        size_t sampleCount = AZStd::min<size_t>(maxSampleCount, m_valueLog.size());
+        float average = 0.f;
+        if (sampleCount > 0)
+        {
+            average = m_valueLog.at(0);
```
Comment (jeremyong-az) [inline Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp:36]: In the STL anyways, the distinction between `at` and `operation[]` is that the former throws an exception for out of bounds access. Is that intended here?

File: Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp [hunk 2]
```diff
@@ -1,105 +1,112 @@
...
+    {
+        AZ_Assert(m_maxSamples >= m_runningAverageSamples, "maxSamples must be larger");
+
+        m_valueLog.reserve(m_maxSamples);
+        m_averageLog.reserve(m_maxSamples);
+    }
+
+    float ImGuiHistogramQueue::UpdateDisplayedValues(AZStd::size_t maxSampleCount, float& minValue, float& maxValue)
+    {
+        size_t sampleCount = AZStd::min<size_t>(maxSampleCount, m_valueLog.size());
```
Comment (jeremyong-az) [inline Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp:32]: I think as of C++17, the template argument `size_t` here should be deducible.

File: Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp [hunk 3]
```diff
@@ -1,105 +1,112 @@
...
+    void ImGuiHistogramQueue::PushValue(float value)
+    {
+        m_samplesSinceLastDisplayUpdate++;
+
+        // Update the log of all values
+        if (m_valueLog.size() == m_maxSamples)
+        {
+            m_valueLog.pop_back();
+        }
+        m_valueLog.insert(m_valueLog.begin(), value);
```
Comment (jeremyong-az) [inline Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp:59]: This is an interesting way to implement a ring buffer (I see that it's carry-over from before, but suggest we actually just use/provide a ring-buffer at some point)

**Review by jeremyong-az — APPROVED** [review]
Was there a reason every line in the files needed to change?

**Review by wintermute-motherbrain — APPROVED** [review]

**Comment by rgba16f** [comment]
> Was there a reason every line in the files needed to change?

It wasn't intentional some tool of mine decided to modify every line ending in the file.

File: Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp (hunk 1 above)
Comment (rgba16f) [reply Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp:36]: The existing code was using `at`, not having a strong reason to change it I left it as is.

File: Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp (hunk 3 above)
Comment (jeremyong-az) [reply Gem/Code/Source/Utils/ImGuiHistogramQueue.cpp:59]: FYI @rgba16f we actually do have an AZStd::ring_buffer if you want to use that instead

**Comment by rgba16f** [comment]
Successful build - https://jenkins-pipeline.agscollab.com/view/Projects/job/o3de-atom-sampleviewer-dev/job/rgba16f%252FHistogramFpsToMs/2/
Successful ASV run - 
image

**Pull request merged**